pkg/trace/api/container_linux.go

```diff
@@ -0,0 +1,60 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016 Datadog, Inc.
+
+package api
+
+import (
+	"context"
+	"net"
+	"net/http"
+	"syscall"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/trace/log"
+	"github.com/DataDog/datadog-agent/pkg/util/containers/v2/metrics"
+)
+
+type ucredKey struct{}
+
+const cacheExpire = 5 * time.Minute
+
+// connContext is a function that injects a Unix Domain Socket's User Credentials into the
+// context.Context object provided. This is useful as the ConnContext member of an http.Server, to
+// provide User Credentials to HTTP handlers.
+func connContext(ctx context.Context, c net.Conn) context.Context {
+	s, ok := c.(*net.UnixConn)
+	if !ok {
+		return ctx
```

Comment:
Do we need to log warnings or anything like that if this happens (or the file error below)? What's the side effect of not reaching the happy-path of this function?

Reply:
The first one is completely normal. I've added a debug log for the second. It *should* never happen, but things that should never happen are the perfect place for debug logs.